test/kubernetes_unittest.cc

```diff
@@ -854,24 +863,378 @@ TEST_F(KubernetesTest, KubernetesUpdater) {
+                {"zone", json::string("TestZone")},
+              })},
+              {"type", json::string("gce_instance")},
+            })},
+            {"nodeName", json::string("TestSpecNodeName")},
+          })},
+          {"version", json::string("TestVersion")},
+        })},
+      })},
+    });
+    EXPECT_EQ(pod_metadata->ToString(),
+              k8s_pod_metadata.metadata->ToString());
+  }
+}
+
+void TestServices(testing::FakeServer& server, MetadataStore& store,
+                  const std::string& services_watch_path) {
+  server.WaitForOneStreamWatcher(services_watch_path);
+
+  Timestamp last_services_timestamp = std::chrono::system_clock::now();
+  for (int i = 0; i < 3; i++) {
     // Send a response to the watcher.
-    server.SendStreamResponse(
-        "/api/v1/pods?fieldSelector=spec.nodeName%3DTestNodeName&watch=true",
-        resp->ToString());
+    json::value resp = json::object({
+      {"type", json::string("ADDED")},
+      {"object", json::object({
+        {"metadata", json::object({
+          {"name", json::string("testname")},
+          {"namespace", json::string("testnamespace")},
+        })}
+      })}
+    });
+    server.SendStreamResponse(services_watch_path, resp->ToString());
+
     // Wait until watcher has processed response (by polling the store).
-    MonitoredResource resource("k8s_pod",
+    MonitoredResource resource("k8s_cluster",
                                {{"cluster_name", "TestClusterName"},
-                                {"location", "TestClusterLocation"},
-                                {"namespace_name", "TestNamespace"},
-                                {"pod_name", "TestPodName"}});
-    EXPECT_TRUE(
-        WaitForNewerCollectionTimestamp(store, resource, last_nodes_timestamp));
+                                {"location", "TestClusterLocation"}});
+    EXPECT_TRUE(WaitForNewerCollectionTimestamp(
+        store, resource, last_services_timestamp));
 
+    // Update last_services_timestamp.
     const auto metadata_map = store.GetMetadataMap();
     const auto& metadata = metadata_map.at(resource);
-    // TODO: Insert tests of metadata values.
-    last_pods_timestamp = metadata.collected_at;
+    last_services_timestamp = metadata.collected_at;
+
+    // Verify values in the store.
+    EXPECT_EQ("TestVersion", metadata.version);
+    EXPECT_FALSE(metadata.is_deleted);
+    EXPECT_EQ(Timestamp(), metadata.created_at);
+    json::value expected_cluster = json::object({
+      {"blobs", json::object({
+        {"services", json::array({
+          json::object({
+            {"api", json::object({
+              {"pods", json::array({})},
+              {"raw", json::object({
+                {"metadata", json::object({
+                  {"name", json::string("testname")},
+                  {"namespace", json::string("testnamespace")},
+                })},
+              })},
+              {"version", json::string("1.6")},  // Hard-coded in kubernetes.cc.
```

Comment:
Should we be consistent about the order of fields? "version" appears at the start of the "api" object elsewhere.